src/EdgeAnalysis.jl

```diff
@@ -490,6 +490,69 @@ end
 
 export calcbranchingnumbers
 
+"""
+Calculate the ratio of net consumption/termination of core radicals by each edge reaction to the net consumption/termination of core radicals by all core reactions
+"""
+function calcradconstermratios(sim,reactantinds,productinds,corespcsinds,corerxninds,edgerxninds,edgereactionrates,coreradicalnetconsumptionrates,corespeciesnetterminationrates)
+    radconstermratios = zeros(length(edgereactionrates))
+    for ind in 1:length(edgereactionrates)
+        index = edgerxninds[ind]
+        reactionrate = edgereactionrates[ind]
+            
+        if reactionrate > 0
+            reactantside = reactantinds[:,index]
+            productside = productinds[:,index]
+        else
+            reactantside = productinds[:,index]
+            productside = reactantinds[:,index]
+        end
+            
+        productrade = [sim.species[i].radicalelectrons for i in productside if i != 0]
+        reactantrade = [sim.species[i].radicalelectrons for i in reactantside if i != 0]
+
+        if length(productrade) == 1 && length(reactantrade) == 2
+            if productrade[1] == 0 && reactantrade[1] == 1 && reactantrade[2] == 1
```

Comment:
Something here productrade and reactantrade  should never be 0 so why do we have checks for that?